### ISSUE TITLE
make imap in bote only bind the configured address and port (Trac #1680)

### ISSUE THREAD

**Issue by str4d** [i2p/i2p.i2p-bote#68]
when testing #67 and https://trac.i2p2.de/ticket/1679 I noticed that the imapserver also binds the default imap address on port 143 which caused a permission error and made my bote hang.

here is the fix:

```text
--- a/src/main/java/i2p/bote/imap/ImapService.java
+++ b/src/main/java/i2p/bote/imap/ImapService.java
@@ -123,6 +123,8 @@ public class ImapService extends IMAPServer {
             // select strong cipher suites
             cfg.setProperty("tls.supportedCipherSuites.cipherSuite",
                     StrongTls.getRecommendedCipherSuites(s.getSupportedCipherSuites()));
+            // set address and port from config in order to avoid binding the default port.
+            cfg.setProperty("bind", configuration.getImapAddress() + ":" + configuration.getImapPort());
         } catch (IOException e) {
             log.error("Couldn't determine supported cipher suites", e);
         } finally {
@@ -133,8 +135,6 @@ public class ImapService extends IMAPServer {
         }
         configure(cfg);   // use the defaults for the rest
 
-        setListenAddresses(new InetSocketAddress(configuration.getImapAddress(), configuration.getImapPort()));
-
         mailboxSessionMapperFactory = new MapperFactory(folderManager);
         MailboxACLResolver aclResolver = createMailboxACLResolver();
         GroupMembershipResolver groupMembershipResolver = new GroupMembershipResolver() {

```

Migrated from https://trac.i2p2.de/ticket/1680
```json
{
    "status": "closed", 
    "changetime": "2016-01-24T20:35:41", 
    "description": "when testing #1678 and #1679 I noticed that the imapserver also binds the default imap address on port 143 which caused a permission error and made my bote hang.\n\nhere is the fix:\n\n{{{\n--- a/src/main/java/i2p/bote/imap/ImapService.java\n+++ b/src/main/java/i2p/bote/imap/ImapService.java\n@@ -123,6 +123,8 @@ public class ImapService extends IMAPServer {\n             // select strong cipher suites\n             cfg.setProperty(\"tls.supportedCipherSuites.cipherSuite\",\n                     StrongTls.getRecommendedCipherSuites(s.getSupportedCipherSuites()));\n+            // set address and port from config in order to avoid binding the default port.\n+            cfg.setProperty(\"bind\", configuration.getImapAddress() + \":\" + configuration.getImapPort());\n         } catch (IOException e) {\n             log.error(\"Couldn't determine supported cipher suites\", e);\n         } finally {\n@@ -133,8 +135,6 @@ public class ImapService extends IMAPServer {\n         }\n         configure(cfg);   // use the defaults for the rest\n \n-        setListenAddresses(new InetSocketAddress(configuration.getImapAddress(), configuration.getImapPort()));\n-\n         mailboxSessionMapperFactory = new MapperFactory(folderManager);\n         MailboxACLResolver aclResolver = createMailboxACLResolver();\n         GroupMembershipResolver groupMembershipResolver = new GroupMembershipResolver() {\n\n}}}\n", 
    "reporter": "kay", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1453667741472454", 
    "component": "apps/plugins", 
    "summary": "make imap in bote only bind the configured address and port", 
    "priority": "minor", 
    "keywords": "I2P-Bote imap", 
    "version": "0.9.22", 
    "parents": "1678", 
    "time": "2015-10-14T17:06:44", 
    "milestone": "0.9.24", 
    "owner": "", 
    "type": "defect"
}
```

**Comment by str4d** [comment]
Trac update at 20151014T17:11:09: kay changed keywords from "I2P-Bote imap" to "I2P-Bote imap needs_review"

**Comment by str4d** [comment]
Trac update at 20151014T17:14:45: kay changed status from "new" to "testing"

**Comment by str4d** [comment]
Trac update at 20151014T21:15:01: 
* str4d commented:

> Please do not change tickets to testing until the fix has actually been implemented. As the description says, testing is for marking the ticket as resolved, fixed and could be closed. If the fix has not been implemented, this is incorrect.


* str4d changed status from "testing" to "needs_work"

**Comment by str4d** [comment]
Trac update at 20151102T12:20:49: str4d changed parents from "1679" to "1678"

**Comment by str4d** [comment]
Trac update at 20160124T20:35:41: 
* str4d commented:

> Fixed in `09503df3afa7b73ba88b2b67667cc81117274a24`.


* str4d changed keywords from "I2P-Bote imap needs_review" to "I2P-Bote imap"
* str4d changed milestone from "undecided" to "0.9.24"
* str4d changed resolution from "" to "fixed"
* str4d changed status from "needs_work" to "closed"
